text/0182-otlp-remote-parent.md

```diff
@@ -0,0 +1,108 @@
+# Export SpanContext.IsRemote in OTLP
+
+Update OTLP to indicate whether a span's parent is remote.
+
+## Motivation
+
+It is sometimes useful to post-process or visualise only entry-point spans: spans which either have no parent (trace roots), or which have a remote parent.
+For example, the Elastic APM solution highlights entry-point spans (Elastic APM refers to these as "transactions") and surfaces these as top-level operations
+in its user interface.
+
+The goal is to identify the spans which represent a request that is entering a service, or originating within a service, without having to first assemble the
+complete distributed trace as a DAG (Directed Acyclic Graph). It is trivially possible to identify trace roots, but it is not possible to identify spans with
+remote parents.
+
+Here is a contrived example distributed trace, with a border added to the entry-point spans:
+
+```mermaid
+graph TD
+    subgraph comments_service
+    POST_comments(POST /comment)
+    POST_comments --> comments_send(comments send)
+    end
+
+    subgraph auth_service
+    POST_comments --> POST_auth(POST /auth)
+    POST_auth --> LDAP
+    end
+
+    subgraph user_details_service
+    POST_comments --> GET_user_details(GET /user_details)
+    GET_user_details --> SELECT_users(SELECT FROM users)
+    end
+
+    subgraph comments_inserter
+    comments_send --> comments_receive(comments receive)
+    comments_receive --> comments_process(comments process)
+    comments_process --> INSERT_comments(INSERT INTO comments)
+    end
+
+    style POST_comments stroke-width:4
+    style POST_auth stroke-width:4
+    style GET_user_details stroke-width:4
+    style comments_receive stroke-width:4
+```
+
+## Explanation
+
+The OTLP encoding for spans has a boolean `parent_span_is_remote` field for identifying whether a span's parent is remote or not.
+All OpenTelemetry SDKs populate this field, and backends may use it to identify a span as being an entry-point span.
+A span can be considered an entry-point span if it has no parent (`parent_span_id` is empty), or if `parent_span_is_remote` is true.
+
+## Internal details
+
+The first part would be to update the trace protobuf, adding a `boolean parent_span_is_remote` field to the
```

Comment:
L48 says "The OTLP encoding for spans has a boolean `parent_span_is_remote` field", what does it refer to if not protobuf?

Reply:
Not sure I understand your question, let's see:

This is the implementation detail of L48. In case it's not clear: L48 is written "as though it was already implemented and you were explaining it to a user", per the template: https://github.com/open-telemetry/oteps/blob/main/0000-template.md#explanation